src/linalg/logdet.jl

```diff
@@ -31,7 +31,9 @@ function LinearAlgebra.logdet(m::LinearMixedModel{T}) where {T}
         s += LD(L[kp1choose2(j)])
     end
     if m.optsum.REML
-        s += LD(L[kp1choose2(nre + 1)])
+        lastL = last(L)
+        s += LD(lastL)  # this adds the log of sqrtpwrss
+        s -= log(last(lastL))
```

Comment:
I don't think it should be too bad here because it is the log of the square root of the penalized residual sum of squares.  It shouldn't be especially large or small.  Also, this block is only used for REML estimation and the default is ML.

An alternative would be to define a function like `LD` that iterates over `diagind(size(d, 1), size(d, 2) - 1)` instead of `diagind(d)`.

Reply:
I think this is fine for now, but if we ever run into weird trouble on REML fits, it'll be the first place I look.